test/make_dist.py

```diff
@@ -52,115 +51,78 @@ def build_dist():
     return subprocess.check_output(["make", "dump-dist"], universal_newlines=True).strip()
 
 
-def download_dist(wait=False):
-    '''Download dists tarball for current git SHA from GitHub
+def download_cache(wait=False):
+    '''Download pre-built webpack for current git SHA from GitHub
 
     These are produced by .github/workflows/build-dist.yml for every PR and push.
     This is a lot faster than having to npm install and run webpack.
 
-    Returns path to downloaded tarball, or None if it isn't available.
-    This can happen because the current directory is not a git checkout, or it is
-    a SHA which is not pushed/PRed.
+    Returns True when successful, or False if the download isn't available.
+    This can happen because dist/ already exists, or the current directory is not a git checkout,
+    or it is a SHA which is not pushed/PRed.
     '''
     try:
         sha = subprocess.check_output(["git", "rev-parse", "HEAD"], stderr=subprocess.DEVNULL).decode().strip()
     except subprocess.CalledProcessError:
         message("make_dist: not a git repository")
-        return None
+        return False
 
     if subprocess.call(["git", "diff", "--quiet", "--", ":^test", ":^packit.yaml", ":^.github"]) > 0:
         message("make_dist: uncommitted local changes, skipping download")
-        return None
+        return False
 
-    dists = glob.glob(f"cockpit-*{sha[:8]}*.tar.xz")
-    if dists:
-        message("make_dist: already downloaded", dists[0])
-        return os.path.abspath(dists[0])
+    dist_git_checkout = os.path.join(os.getenv("XDG_CACHE_HOME", os.path.expanduser("~/.cache")), "cockpit-dev", CACHE_REPO)
+
+    if not os.path.exists(dist_git_checkout):
+        message(f"make_dist: Creating dist cache {dist_git_checkout}")
+        os.makedirs(dist_git_checkout)
+        subprocess.check_call(["git", "-C", dist_git_checkout, "init", "--bare", "--quiet"])
```

Comment:
It probably makes more sense to use `--git-dir` instead of `-C`.  Also: consider passing the directory as an argument to `init` rather.

Reply:
Thanks! Done.